### PR TITLE
⚡️ Modernize

### PR DESCRIPTION
**Pull request by nettofarah**
## What this PR does
- [x] Migrate main codebase to TypeScript
- [x] Update dependencies
- [x] Migrate tests from `ava` to `jest`
- [x] Update demos
- [x] Upgrade Storybook
- [x] Publish type definitions
- [x] Move from `xo` to `eslint`
- [x] Remove dependency on Babel
- [x] Migrate from `inferno` to `preact` (so we can get support for hooks)

## Highlights ⚡️
- Demos now include source code: https://segmentio.github.io/consent-manager/
image

- Bundle size reduction from 58k to 44k for the standalone script tag

- Type definitions and type safe code

- Replaces Inferno with Preact, which will allow for us to use more modern features of react

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "compilerOptions": {
```
Comment (Rowno) [inline tsconfig.json]: Might want to include `"declaration": true`

File: package.json [hunk 2]
```diff
@@ -19,22 +19,21 @@
     "standalone"
   ],
   "scripts": {
-    "test": "xo && ava",
+    "test": "jest",
```
Comment (Rowno) [inline package.json]: Are you going to replace `xo` with `eslint`?

File: src/types.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+export type WindowWithAJS = Window &
+  typeof globalThis & {
+    // TODO: add AJS types
+    analytics?: any
```
Comment (Rowno) [inline src/types.ts]: `@types/segment-analytics` 😉

File: package.json (hunk 2 above)
Comment (nettofarah) [reply package.json]: Yes!

File: tsconfig.json (hunk 1 above)
Comment (nettofarah) [reply tsconfig.json]: I will. I removed as you can't use `declaration` with `allowJs` (and I wasn't done migrating everything to TS yet)

File: src/types.ts (hunk 3 above)
Comment (nettofarah) [reply src/types.ts]: Yes! coming :)

File: webpack.config.js [hunk 4]
```diff
@@ -5,30 +5,26 @@ const pkg = require('./package.json')
...
     path: path.join(__dirname, 'standalone'),
     filename: 'consent-manager.js',
     library: 'consentManager'
   },
   resolve: {
     alias: {
-      react: 'inferno-compat',
-      'react-dom': 'inferno-compat',
+      react: 'preact-compat',
+      'react-dom': 'preact-compat',
```
Comment (Rowno) [inline webpack.config.js]: Just FYI that I ran into a bunch of random edge issues when trying to use Preact, which is why I went with Inferno instead.

Comment (nettofarah) [reply webpack.config.js]: I remember that. Things look fairly safe so far (based on my manual testing), but I'll keep QAing everything before releasing something new.

File: tsconfig.json (hunk 1 above)
Comment (Rowno) [reply tsconfig.json]: You can with TypeScript 3.7 😉 https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/

Comment (nettofarah) [reply tsconfig.json]: One step at a time, Rolee 😅

**Pull request merged**